### PR TITLE
Refactor commands to az aosm nfd|nsd build

### PR DESCRIPTION
**Pull request by sunnycarter**
I want to get these in before I send the commands to the CLI review.

Others should be aware of these changes to avoid merge problems.

### CODE REVIEW

**Comment by jddarby** [comment]
I think we said at some point that having the `--publish` in the build command was maybe not a good idea and we should be encouraging users to check the autogenerated definitions before they publish. Is this something we want to do in this PR?

**Comment by jddarby** [comment]
Are print statements the correct way of logging? Did you ever find out?

**Comment by sunnycarter** [comment]
> Are print statements the correct way of logging? Did you ever find out?

Yes, it seems that print is the right way to output to screen. I tried to also debug log these when I did my VNF stuff, which was a pain.

File: src/aosm/azext_aosm/_params.py [hunk 1]
```diff
@@ -71,9 +71,28 @@ def load_arguments(self: AzCommandsLoader, _):
             help="Optional path to a parameters file for the manifest definition file. Use to override publish of the built definition and config with alternative parameters.",
         )
 
-    with self.argument_context("aosm generate-config") as c:
```
[the remark this answers is not in the document]
Comment (sunnycarter) [reply src/aosm/azext_aosm/_params.py:74]: was wrong before and I've deleted it

**Comment by sunnycarter** [comment]
> I think we said at some point that having the `--publish` in the build command was maybe not a good idea and we should be encouraging users to check the autogenerated definitions before they publish. Is this something we want to do in this PR?

I have done that now

**Pull request merged**